### PR TITLE
Don't modify ClientOptions; Fixed tests to not set both host and environment

### PR DESCRIPTION
**Pull request by trenouf**

### CODE REVIEW

**Comment by trenouf** [comment]
@paddy Please review and merge (or tell me I can merge). Ta.

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeConnectFailTest.java [hunk 1]
```diff
@@ -165,8 +168,9 @@ public void connect_while_disconnected() {
 	/**
 	 * Verify that the connection enters the disconnected state, after a token
 	 * used for successful connection expires
+	 * DISABLED: See issue https://github.com/ably/ably-java/issues/183
 	 */
-	@Test
+	/*@Test*/
```
Comment (paddybyers) [inline lib/src/test/java/io/ably/lib/test/realtime/RealtimeConnectFailTest.java]: TBH I would prefer not to mix too many issues into the same PR. Can we keep this to the changes required for the issue stated in the title (or change the title)?

**Comment by trenouf** [comment]
@paddy No unexpected failures on travis. Please review'n'merge.

**Comment by paddybyers** [comment]
LGTM, thanks

**Pull request merged**